frame/balances/src/lib.rs

```diff
@@ -1209,6 +1196,44 @@ where
 	}
 }
 
+impl<T: Trait<I>, I: Instance> VestingCurrency<T::AccountId> for Module<T, I>
+where
+	T::Balance: MaybeSerializeDeserialize + Debug
+{
+	type Moment = T::BlockNumber;
+
+	/// Get the amount that is currently being vested and cannot be transferred out of this account.
+	fn vesting_balance(who: &T::AccountId) -> T::Balance {
+		if let Some(v) = Self::vesting(who) {
+			Self::free_balance(who)
+				.min(v.locked_at(<system::Module<T>>::block_number()))
+		} else {
+			Zero::zero()
+		}
+	}
+
+	/// Insert a vesting schedule for a given account.
+	/// Will overwrite any existing schedule.
+	fn insert_vesting_schedule(
+		who: &T::AccountId,
+		locked: T::Balance,
+		per_block: T::Balance,
+		starting_block: T::BlockNumber
+	) {
+		let vesting_schedule = VestingSchedule {
+			locked,
+			per_block,
+			starting_block
+		};
+		<Vesting<T, I>>::insert(who, vesting_schedule);
```

Comment:
this would clobber any pre-existing vesting schedule. it should probably return a result and fail if there already exists one.

Reply:
Updated to `add_vesting_schedule`, which returns `Err` if `Vesting::exists(who)`.